ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -270,41 +278,56 @@ type CardanoHardForkConstraints c =
-          PCons translateChainDepStateByronToShelleyWrapper
-        $ PCons translateChainDepStateAcrossShelley
-        $ PCons translateChainDepStateAcrossShelley
-        $ PCons translateChainDepStateAcrossShelley
-        $ PNil
-    , translateLedgerView    =
-          PCons translateLedgerViewByronToShelleyWrapper
-        $ PCons translateLedgerViewAcrossShelley
-        $ PCons translateLedgerViewAcrossShelley
-        $ PCons translateLedgerViewAcrossShelley
-        $ PNil
-    }
+          translateLedgerState   =
+              PCons translateLedgerStateByronToShelleyWrapper
+            $ PCons translateLedgerStateShelleyToAllegraWrapper
+            $ PCons translateLedgerStateAllegraToMaryWrapper
+            $ PCons translateLedgerStateMaryToAlonzoWrapper
+            $ PCons translateLedgerStateAlonzoToBabbageWrapper
+            $ PNil
+        , translateChainDepState =
+              PCons translateChainDepStateByronToShelleyWrapper
+            $ PCons translateChainDepStateAcrossShelley
+            $ PCons translateChainDepStateAcrossShelley
+            $ PCons translateChainDepStateAcrossShelley
+            $ PCons translateChainDepStateAcrossShelley
+            $ PNil
+        , translateLedgerView    =
+              PCons translateLedgerViewByronToShelleyWrapper
+            $ PCons translateLedgerViewAcrossShelley
+            $ PCons translateLedgerViewAcrossShelley
+            $ PCons translateLedgerViewAcrossShelley
+            $ PCons translateLedgerViewAcrossShelley
+            $ PNil
+        }
   hardForkChainSel =
         -- Byron <-> Shelley, ...
         TCons (   CompareBlockNo
-               :* CompareBlockNo
-               :* CompareBlockNo
-               :* CompareBlockNo
-               :* Nil)
+              :* CompareBlockNo
+              :* CompareBlockNo
+              :* CompareBlockNo
+              :* CompareBlockNo
+              :* Nil)
         -- Shelley <-> Allegra, ...
-      $ TCons (SelectSameProtocol :* SelectSameProtocol :* SelectSameProtocol :* Nil)
+      $ TCons (SelectSameProtocol :* SelectSameProtocol :* SelectSameProtocol :* alonzoBabbageEraSelection :* Nil)
         -- Allegra <-> Mary, ...
-      $ TCons (SelectSameProtocol :* SelectSameProtocol :* Nil)
+      $ TCons (SelectSameProtocol :* SelectSameProtocol :* alonzoBabbageEraSelection :* Nil)
         -- Mary <-> Alonzo, ...
-      $ TCons (SelectSameProtocol :* Nil)
-        -- Alonzo <-> ...
+      $ TCons (SelectSameProtocol :* alonzoBabbageEraSelection :* Nil)
+        -- Alonzo <-> Babbage, ...
+      $ TCons (alonzoBabbageEraSelection :* Nil)
+        -- Babbage <-> ...
       $ TCons Nil
       $ TNil
+    where
+      -- Across alonzo and Babbage, the protocol changes, but the select view
+      -- remains the same.
+      alonzoBabbageEraSelection :: AcrossEraSelection
+        (ShelleyBlock (TPraos c) e1)
+        (ShelleyBlock (Praos c) e2)
+      alonzoBabbageEraSelection = CustomChainSel (
```

Comment:
Harumph. Should `SelectSameProtocol` itself accomodate this? Or a similar constructor next to it? I'm sad to introduce a use of `CustomChainSel` that is so thin.

Reply:
Clarke pointed out that `SelectSameProtocol` has Haddock that explicitly says we intentionally don't accept structural equivalence, only nominal. So using `CustomChainSel` is probably right here. Also, we don't think doing so has any unexpected consequences, so this seems fine as-is.